TestHttpsInsecure failures on certain darwin builders

We're getting `TestHttpsInsecure` failures on some (but not all) of the darwin builders.

OSX 10.11 and 10.12 are fine.
The OSX 10.8 and 10.10 builders fail. The darwin-arm-a1549ios builder is also failing.

This is the OSX error:

```
--- FAIL: TestHttpsInsecure (10.04s)
	fetch_test.go:305: grabProfile(https+insecure://127.0.0.1:59997/debug/pprof/profile) got PeriodType: cpu nanoseconds
		Period: 10000000
		Time: 2017-06-20 11:36:19.953808169 -0700 PDT
		Duration: 10.0
		Samples:
		samples/count cpu/nanoseconds
		        492 4920000000: 1 2 3 4 5 6 
		Locations
		     1: 0x105efd3 M=1 runtime.kevent /var/folders/6g/8k13h11n2zb8hxx99wk6f6y40000gn/T/workdir/go/src/runtime/sys_darwin_amd64.s:526 s=0
		     2: 0x102a672 M=1 runtime.netpoll /var/folders/6g/8k13h11n2zb8hxx99wk6f6y40000gn/T/workdir/go/src/runtime/netpoll_kqueue.go:79 s=0
		     3: 0x103405a M=1 runtime.findrunnable /var/folders/6g/8k13h11n2zb8hxx99wk6f6y40000gn/T/workdir/go/src/runtime/proc.go:2099 s=0
		     4: 0x10349eb M=1 runtime.schedule /var/folders/6g/8k13h11n2zb8hxx99wk6f6y40000gn/T/workdir/go/src/runtime/proc.go:2237 s=0
		     5: 0x1034d05 M=1 runtime.park_m /var/folders/6g/8k13h11n2zb8hxx99wk6f6y40000gn/T/workdir/go/src/runtime/proc.go:2300 s=0
		     6: 0x105aeda M=1 runtime.mcall /var/folders/6g/8k13h11n2zb8hxx99wk6f6y40000gn/T/workdir/go/src/runtime/asm_amd64.s:286 s=0
		Mappings
		1: 0x0/0x0/0x0   [FN]
		, want function "TestHttpsInsecure"
FAIL
FAIL	cmd/vendor/github.com/google/pprof/internal/driver	10.545s
```

Full logs: [OSX 10.10](https://build.golang.org/log/bfb7c4db36ddd4a4f801d5b33cfe8088b1a80c73), [OSX 10.8](https://build.golang.org/log/e9207dc8ed9dd01f11938974217b26bf2b901715), [iphone6](https://build.golang.org/log/e4a96f0bbced472564e20e3258a7458d4a0a0859). The last one is a little different.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,14 +1,33 @@
 language: go
-go:
-  - 1.7.x
-  - 1.8.x
-  - master
 
 go_import_path: github.com/google/pprof
 
-os:
-  - linux
-  - osx
+matrix:
+  include:
+    - os: linux
+      go: 1.7.x
+    - os: linux
+      go: 1.8.x
+    - os: linux
+      go: master
+    - os: osx
+      osx_image: xcode6.4
+      go: 1.8.x
+    - os: osx
+      osx_image: xcode6.4
+      go: master
+    - os: osx
+      osx_image: xcode7.3
+      go: 1.8.x
+    - os: osx
+      osx_image: xcode7.3
+      go: master
+    - os: osx
+      osx_image: xcode8.3
+      go: 1.8.x
+    - os: osx
+      osx_image: xcode8.3
+      go: master
 
 before_install:
   - go get -u github.com/golang/lint/golint honnef.co/go/tools/cmd/...

--- a/internal/driver/fetch_test.go
+++ b/internal/driver/fetch_test.go
@@ -381,7 +381,9 @@ func TestHttpsInsecure(t *testing.T) {
 	go func() {
 		deadline := time.Now().Add(5 * time.Second)
 		for time.Now().Before(deadline) {
-			// Simulate a hotspot function.
+			// Simulate a hotspot function. Spin in the inner loop for 100M iterations
+			// to ensure we get most of the samples landed here rather than in the
+			// library calls. We assume Go compiler won't elide the empty loop.
 			for i := 0; i < 1e8; i++ {
 			}
 			runtime.Gosched()

--- a/internal/driver/fetch_test.go
+++ b/internal/driver/fetch_test.go
@@ -382,6 +382,9 @@ func TestHttpsInsecure(t *testing.T) {
 		deadline := time.Now().Add(5 * time.Second)
 		for time.Now().Before(deadline) {
 			// Simulate a hotspot function.
+			for i := 0; i < 1e8; i++ {
+			}
+			runtime.Gosched()
 		}
 	}()
 
@@ -409,11 +412,22 @@ func TestHttpsInsecure(t *testing.T) {
 		t.Fatal(err)
 	}
 	if len(p.SampleType) == 0 {
-		t.Fatalf("grabProfile(%s) got empty profile: len(p.SampleType)==0", address)
+		t.Fatalf("fetchProfiles(%s) got empty profile: len(p.SampleType)==0", address)
 	}
-	if err := checkProfileHasFunction(p, "TestHttpsInsecure"); err != nil {
-		t.Fatalf("grabProfile(%s) %v", address, err)
+	if len(p.Function) == 0 {
+		t.Fatalf("fetchProfiles(%s) got non-symbolized profile: len(p.Function)==0", address)
 	}
+	if err := checkProfileHasFunction(p, "TestHttpsInsecure"); !badSigprofOS[runtime.GOOS] && err != nil {
+		t.Fatalf("fetchProfiles(%s) %v", address, err)
+	}
+}
+
+// Some operating systems don't trigger the profiling signal right.
+// See https://github.com/golang/go/issues/13841.
+var badSigprofOS = map[string]bool{
+	"darwin": true,
+	"netbsd": true,
+	"plan9":  true,
 }
 
 func checkProfileHasFunction(p *profile.Profile, fname string) error {

